avatars/processors/proportions.py

```diff
@@ -88,4 +91,11 @@ def postprocess(self, source: pd.DataFrame, dest: pd.DataFrame) -> pd.DataFrame:
             sub_df = dest[self.variable_names].mul(dest[self.reference], axis=0)
         dest = dest.drop(columns=self.variable_names)
         dest = pd.concat([dest, sub_df], axis=1)[col_order]
+
+        # Perform rounding of the postprocess variable to the expected number of decimals.
+        # We use saferounding here to force the sum of rounded variables to remain unchanged. 
+        if self.sum_to_one:
+            for i, row in enumerate(dest[self.variable_names].values):
+                if not np.any(np.isnan(row)):
+                    dest.loc[i, self.variable_names] = saferound(row, self.decimal_count)
         return dest
```

Comment:
```suggestion
        if not self.sum_to_one:
            return dest
        
        for i, row in enumerate(dest[self.variable_names].values):
            if not np.any(np.isnan(row)):
                dest.loc[i, self.variable_names] = saferound(row, self.decimal_count)
        return dest
```

nitpick: early return leads to less indentation